src/tools/linker/.editorconfig

```diff
@@ -0,0 +1,11 @@
+### Code Style Analyzers
```

Comment:
Can we delete this and use repo-wide global configurations (`CodeAnalysis.{src,test}.globalconfig`)?

Reply:
This already takes into account the CodeAnalysis.{src,test}.globalconfig. This is an override of what's in that file. For example, IDE0005 is not enabled globally because a lot of the files in the runtime have `#ifdef` that could incorrectly flag unnecessary usings/imports, but in the case of linker we use that rule. The original editorconfig from linker is actually more comprehensive than this file, but since is inheriting most from the globalconfig these were the only remaining items